Truncate MAVLink 2 messages

Erk- opened the pull request:
The MAVLink documentation states the following[^1]:
> *MAVLink 2* implementations *must* truncate any empty (zero-filled)
> bytes at the end of the serialized payload before it is sent.

So this PR implements that.

[^1]: https://mavlink.io/en/guide/serialization.html#payload_truncation

patrickelectric reviewed (commented):
Please squash the commits, I'll do some tests to validate the PR just to make sure.

Erk- commented:
I have rebased and squashed it, but it will likely need some minor changes when #124 is merged similar to the ones I have [here](https://github.com/fusion-engineering-forks/rust-mavlink/blob/fusion/mavlink-changes/build/util.rs#L22)

patrickelectric commented on src/lib.rs:
```diff
@@ -71,7 +71,8 @@ where
     fn message_id(&self) -> u32;
     fn message_name(&self) -> &'static str;
-    fn ser(&self) -> Vec<u8>;
+    fn ser_v1(&self) -> Vec<u8>;
```
Why not creating a `ser` function that receives as argument `MavlinkVersion` ? Like what we have for `deser`. The only difference of `ser_v1` and `ser_v2` is the trailing removal logic.

patrickelectric commented on build/util.rs:
```diff
@@ -9,3 +12,18 @@ pub fn to_module_name<P: Into<PathBuf>>(file_name: P) -> String {
+/// zero byte.
+pub fn q_remove_trailing_zeroes(buf: proc_macro2::Ident) -> TokenStream {
+    quote! {
+        while let Some(&0) = #buf.last() {
```
Doing so we end up duplicating the same code again and again inside the binary, it's better to have a global function that abstract such behavior over having the same logic around every serialized message.

patrickelectric replied:
To do that, we probably need to move this function to a library that will be used by all dialects.

patrickelectric commented on build/util.rs:
```diff
@@ -9,3 +12,18 @@ pub fn to_module_name<P: Into<PathBuf>>(file_name: P) -> String {
+///
+/// There must always be at least one remaining byte even if it is a
+/// zero byte.
+pub fn q_remove_trailing_zeroes(buf: proc_macro2::Ident) -> TokenStream {
```
What is the reason behind the q prefix ? That sounds unnecessary.

patrickelectric reviewed (approved):
Thanks!
